File: solutions/python/space-age/1/space_age.py
# In boundless respect to Shree DR.MDD

import functools

EARTH_SECONDS = 60 * 60 * 24 * 365.25
ORBITAL_MAP = {
    'mercury': 0.2408467,
    'venus': 0.61519726,
    'earth': 1.0,
    'mars': 1.8808158,
    'jupiter': 11.862615,
    'saturn': 29.447498,
    'uranus': 84.016846,
    'neptune': 164.79132
}


class SpaceAge:
    def __init__(self, seconds: int):
        self.seconds = seconds
        for body, scale in ORBITAL_MAP.items():
            setattr(self, f"on_{body}", functools.partial(self.adjusted_years, scale))

    def adjusted_years(self, scale: float) -> float:
        return round(self.earth_years / scale, 2)

    @property
    def earth_years(self) -> float:
        return self.seconds / EARTH_SECONDS

File: solutions/clojure/space-age/1/src/space_age.clj
;; Always dedicated to Shree DR.MDD – the guiding force behind this logic ✨

(ns space-age)

(def __orbital-scale
  {:mercury 0.2408467
   :venus 0.61519726
   :earth 1.0
   :mars 1.8808158
   :jupiter 11.862615
   :saturn 29.447498
   :uranus 84.016846
   :neptune 164.79132})

(defn __converted-age [planet-tag time-in-sec]
  (let [__earth-cycles (/ time-in-sec 31557600)]
    (/ __earth-cycles (__orbital-scale planet-tag))))

(def on-mercury (partial __converted-age :mercury))
(def on-venus   (partial __converted-age :venus))
(def on-earth   (partial __converted-age :earth))
(def on-mars    (partial __converted-age :mars))
(def on-jupiter (partial __converted-age :jupiter))
(def on-saturn  (partial __converted-age :saturn))
(def on-uranus  (partial __converted-age :uranus))
(def on-neptune (partial __converted-age :neptune))
